onnxruntime/core/framework/utils.cc

```diff
@@ -502,6 +530,71 @@ static common::Status ExecuteGraphImpl(const SessionState& session_state,
   return Status::OK();
 }
 
+#ifdef ENABLE_TRAINING
+static common::Status ExecuteGraphImpl(const SessionState& session_state,
```

Comment:
Readability nit: Can you call this `PartialExecuteGraphImpl()` instead of overloading the existing `ExecuteGraphImpl()`? I think this will make the code easier to follow.

(Side-note: is there a particular need for splitting `ExecuteGraph()` and `ExecuteGraphImpl()`? I realize it follows the existing pattern, but it doesn't make a lot of sense to me.)

Reply:
ExecuteGraphImpl is used by ExecuteGraph and ExecuteSubgraph (these were adjacent but ExecutePartialGraph has been inserted between them) as they have slightly different semantics.